src/utils.jl

```diff
@@ -169,6 +169,16 @@ for (type, _, _) in ABSTRACT_QUANTITY_TYPES
 end
 Base.keys(q::UnionAbstractQuantity) = keys(ustrip(q))
 
+# If atol specified in kwargs, validate its dimensions and then strip units
+function _strip_kws(dimcheck, kws)
+    if :atol in keys(kws)
+        dimension(dimcheck) == dimension(kws[:atol]) || throw(DimensionError(dimcheck, kws[:atol]))
+        kws = Dict{Symbol,Any}(kws)
```

Comment:
Wont this create extra allocations? i think it should be a named tuple like `kws = (; kws…, atol=ustrip(kws[:atol]))`

Reply:
I can take a look at that idea. I ran into two issues here that led to this version:
- The `kws` produced when args are passed is immutable so it couldn’t be changed in-place
- If e.g. atol is the only keyword arg passed, then `kws` is a set of Pair’s `Symbol => Quantity`. I tried running a replace operation here instead but found that the ustrip result was by default promoted to a dimensionless Quantity, which generates an error.

Reply:
NamedTuples let you change at construction time. Whatever is given last will take priority in case of duplicate keys:

```julia
julia> kws = (;atol=1.0)
(atol = 1.0,)

julia> (; kws..., atol=2.0)
(atol = 2.0,)
```

And there are zero allocations here.

Reply:
Oh, nice! Wish I had known about this as it would’ve saved me a good bit of troubleshooting. As long as the Pair types change to prevent the stripped result from promoting back into a Quantity it should work great.

Reply:
Just commit an update with these comments addressed.